### PR TITLE
SONARJAVA-1468 @java.lang.Override recognized as Override annotation

### PR DESCRIPTION
**Pull request by benzonico**

### CODE REVIEW

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 1]
```diff
@@ -289,16 +290,23 @@ private boolean isPublic() {
 
   public boolean isAnnotatedOverride() {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: MAJOR The Cyclomatic Complexity of this method "isAnnotatedOverride" is 11 which is greater than 10 authorized. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 2]
```diff
@@ -289,16 +290,23 @@ private boolean isPublic() {
...
     for (AnnotationTree annotationTree : modifiers.annotations()) {
-      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER)) {
-        IdentifierTree identifier = (IdentifierTree) annotationTree.annotationType();
-        if (Override.class.getSimpleName().equals(identifier.name())) {
-          return true;
+      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER) && isOverride((IdentifierTree) annotationTree.annotationType())) {
+        return true;
+      } else if (annotationTree.annotationType().is(Kind.MEMBER_SELECT)) {
+        MemberSelectExpressionTree mse = (MemberSelectExpressionTree) annotationTree.annotationType();
```
Comment (m-g-sonar) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: handling of `MemberSelect` could be extracted to a dedicated method.

File: java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java [hunk 3]
```diff
@@ -289,16 +290,29 @@ private boolean isPublic() {
...
   public boolean isAnnotatedOverride() {
     for (AnnotationTree annotationTree : modifiers.annotations()) {
-      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER)) {
-        IdentifierTree identifier = (IdentifierTree) annotationTree.annotationType();
-        if (Override.class.getSimpleName().equals(identifier.name())) {
-          return true;
-        }
+      if (annotationTree.annotationType().is(Tree.Kind.IDENTIFIER) && isOverride((IdentifierTree) annotationTree.annotationType())) {
+        return true;
+      } else if (annotationTree.annotationType().is(Kind.MEMBER_SELECT) && isJavaLangOverride((MemberSelectExpressionTree) annotationTree.annotationType())) {
```
Comment (SonarTech) [inline java-squid/src/main/java/org/sonar/java/model/declaration/MethodTreeImpl.java]: MAJOR This branch's code block is the same as the block for the branch on line 293. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Pull request merged**